tests/unit/test_signature.cpp

```diff
@@ -0,0 +1,427 @@
+
+    // Verification context considering different padding mode is set.
+    verificationContext.setKey(_keyPair);
+    verificationContext.setPadding(openssl::RSAPaddingMode::PKCS1);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different hashing is set.
+    verificationContext.setPadding(openssl::RSAPaddingMode::PSS);
+    verificationContext.setHashingFunction(openssl::DigestTypes::SHA512);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different masking is set.
+    verificationContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    verificationContext.setMaskingFunction(openssl::DigestTypes::SHA512);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different salt length is set.
+    verificationContext.setMaskingFunction(openssl::DigestTypes::SHA256);
+    verificationContext.setSaltLength(2);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+}
+
+// Signature Failure cases
+
+/**
+ * @brief Unsuccessful case where the message digest length is different of padding mode,
+ *        considering the usage of PKCS1.
+ */
+TEST_F(SignatureTest, testSignatureInvalidDigestLengthPKCS1)
+{
+    std::vector<uint8_t> signature;
+    std::vector<uint8_t> messageDigest;
+    SignatureUtils::Context<AsymmetricPrivateKey> signatureContext(_keyPair);
+
+    messageDigest = sha256(reinterpret_cast<const uint8_t *>(_testMessage.c_str()),
+                           reinterpret_cast<size_t>(_testMessage.length()));
+
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA512);
+    signatureContext.setPadding(openssl::RSAPaddingMode::PKCS1);
+
+    EXPECT_THROW(signature = SignatureUtils::create(signatureContext, messageDigest),
+                 MoCOCrWException);
+}
+
+/**
+ *  @brief Signature fails due to an invalid salt length value.
+ */
+TEST_F(SignatureTest, testSignatureInvalidSaltLength)
+{
+    SignatureUtils::Context<AsymmetricPrivateKey> signatureContext(_keyPair);
+    SignatureUtils::Context<AsymmetricPublicKey> verificationContext(_keyPair);
+
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    signatureContext.setPadding(openssl::RSAPaddingMode::PSS);
+
+    /* Use default salt length */
+    ASSERT_NO_THROW(SignatureUtils::create(signatureContext, _testMessage));
+
+    signatureContext.setSaltLength(200);
+    ASSERT_NO_THROW(SignatureUtils::create(signatureContext, _testMessage));
+
+    signatureContext.setSaltLength(0);
+    ASSERT_NO_THROW(SignatureUtils::create(signatureContext, _testMessage));
+
+    signatureContext.setSaltLength(-1);
```

Comment:
but it might be a good idea to drop a comment here why -1 and -2 are valid (special cases).